lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -795,7 +795,41 @@ def test_render_html_view_with_preview_mode_when_user_already_has_cert(self):
         self.assertIn('course_title_0', response.content)
         self.assertIn('Signatory_Title 0', response.content)
 
-    @freeze_time('2017-09-10 00:00:00Z')
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    @ddt.data(
+        (-2, True),
+        (-2, False),
+        (10, False)
+    )
+    @ddt.unpack
+    def test_html_view_certificate_availability_date_for_instructor_paced_courses(self, cert_avail_delta, self_paced):
+        """
+        test certificate web view should display the certificate availability date
+        as the issued date for instructor-paced courses
+        """
+        self.course.self_paced = self_paced
+        self.course.certificate_available_date = datetime.datetime.today() + datetime.timedelta(cert_avail_delta)
+        self.course.save()
+        self._add_course_certificates(count=1, signatory_count=1, is_active=True)
+        test_url = get_certificate_url(
+            user_id=self.user.id,
+            course_id=unicode(self.course.id)
+        )
+
+        if self_paced or self.course.certificate_available_date > datetime.datetime.today():
+            expected_date = datetime.datetime.today()
```

Comment:
There is still a (maybe not as often though) race condition here - it's possible the day changed between L811 and L819 while this test is running.  To safeguard against this, store the response to the call above in a temporary variable.

Reply:
Good catch, thank you!